src/vcf/assembly_checker.cpp

```diff
@@ -36,21 +40,94 @@ namespace ebi
           ebi::vcf::check_readability_of_file(file_extension);
 
           if (file_extension == NO_EXT) {
-              return process_vcf_ref(vcf_input, fasta_input, fasta_index_input, assembly_report, outputs);
+              return process_vcf_ref(vcf_input, fasta, assembly_report, outputs);
           } else {
               boost::iostreams::filtering_istream uncompressed_input;
               ebi::vcf::create_uncompressed_stream(vcf_input, file_extension, uncompressed_input);
-              return process_vcf_ref(uncompressed_input, fasta_input, fasta_index_input, assembly_report, outputs);
+              return process_vcf_ref(uncompressed_input, fasta, assembly_report, outputs);
           }
       }
 
       bool process_vcf_ref(std::istream & vcf_input,
-                              std::istream & fasta_input,
-                              std::istream & fasta_index_input,
-                              const std::string & assembly_report,
-                              std::vector<std::unique_ptr<ebi::vcf::AssemblyCheckReportWriter>> & outputs)
+                           std::shared_ptr<ebi::vcf::fasta::IFasta> & fasta,
+                           const std::string & assembly_report,
+                           std::vector<std::unique_ptr<ebi::vcf::AssemblyCheckReportWriter>> & outputs) {
+        std::vector<char> vector_line;
+        vector_line.reserve(default_line_buffer_size);
 
-      {
+        if (!(fasta.get())) { // No local/remote fasta file provided
+          std::string reference_remote_location;
+          std::string reference_accession;
+          std::vector<std::string> contigs;
+
+          for (size_t line_num = 1; util::readline(vcf_input, vector_line).size() != 0; ++line_num) {
+            std::string line{vector_line.begin(), vector_line.end()};
+            if (!boost::starts_with(line, "##")) {
+              break;
+            }
+            for (auto &output : outputs) {
+              output->write_meta_info(line);
```

Comment:
This is to ensure that the current metadata line is written to outputs. Otherwise the stream has be "rewinded" back to the beginning. On some platforms random seek may not be available for the input. I had to revert it back in change 880f5a9 .